app/views/contexts/summary/question.py

```diff
@@ -31,54 +25,55 @@ def __init__(
         self,
         question_schema: QuestionSchemaType,
         *,
-        answer_store: AnswerStore,
-        list_store: ListStore,
-        progress_store: ProgressStore,
-        supplementary_data_store: SupplementaryDataStore,
+        data_stores: DataStores,
         schema: QuestionnaireSchema,
-        rule_evaluator: RuleEvaluator,
-        value_source_resolver: ValueSourceResolver,
         location: LocationType,
         block_id: str,
         return_to: Optional[str],
         return_to_block_id: Optional[str] = None,
         return_to_list_item_id: Optional[str] = None,
-        metadata: MetadataProxy | None,
-        response_metadata: MutableMapping,
         language: str,
     ) -> None:
         self.list_item_id = location.list_item_id if location else None
         self.id = question_schema["id"]
         self.type = question_schema["type"]
         self.schema = schema
         self.answer_schemas = iter(question_schema.get("answers", []))
-        self.answer_store = answer_store
-        self.list_store = list_store
-        self.progress_store = progress_store
-        self.supplementary_data_store = supplementary_data_store
+        self.answer_store = data_stores.answer_store
+        self.list_store = data_stores.list_store
+        self.progress_store = data_stores.progress_store
+        self.supplementary_data_store = data_stores.supplementary_data_store
```

Comment:
the only place these attributes are used is in creating `PlaceholderRenderer` on line 305 
they could be removed and the datastores method can be used instead to initialise it

Reply:
I don't think this one is possible as we initialise `PlaceholderRenderer` within the `DataStores` class. So can we pass `DataStores` instance inside itself?
If we can do:
```python
    def placeholder_renderer(
        self,
        language: str,
        schema: QuestionnaireSchema,
        location: LocationType | None = None,
        placeholder_preview_mode: bool | None = False,
    ) -> placeholder_renderer.PlaceholderRenderer:
        return placeholder_renderer.PlaceholderRenderer(
            language=language,
            data_stores=self,
            schema=schema,
            location=location,
            placeholder_preview_mode=placeholder_preview_mode,
        )
```
then I will refactor it as well.

Reply:
I see what you mean with the other ones, makes sense to keep params if you've got those creation methods in data stores
I think this one can be fixed though if you use `placeholder_renderer = self.data_stores.placeholder_renderer` instead of initialising directly?

Reply:
I've just checked/refactored a lot and it seems impossible because of all the circular imports issues for type hints.

Reply:
> I think this one can be fixed though if you use `placeholder_renderer = self.data_stores.placeholder_renderer` instead of initialising directly?

Yeah, this should be possible to change.